pkg/platform/runtime/setup/setup.go

```diff
@@ -579,10 +579,37 @@ func (s *Setup) downloadArtifactWithProgress(unsignedURI string, targetFile stri
 	return nil
 }
 
+// verifyArtifact verifies the checksum of the downloaded artifact matches the checksum given by the
+// platform, and returns an error if the verification fails.
+func (s *Setup) verifyArtifact(archivePath string, a artifact.ArtifactDownload) error {
+	if a.Checksum != "" {
+		logging.Debug("Validating checksum for %s", archivePath)
+	} else {
+		logging.Debug("Skipping checksum validation for %s because the Platform did not provide a checksum to validate against.")
+		return nil
+	}
+
+	checksum, err := fileutils.Hash(archivePath)
```

Comment:
We should maybe rename `Hash` to `Sha256Hash` or something more explicit. Since this usage cares strongly about the hashing mechanism used, whereas that function does not. Seems we could implicitly break runtime checksum validation by making changes to `fileutils.Hash()` without realizing that it's used in such opinionated ways.